### DISCUSSION TITLE
Error in the source for `sensor.export_yield_sale_2`

### DISCUSSION THREAD

**Discussion by JensenNick** [JensenNick/huawei_solar_pees#12] [Forum]
Hi all. Sorry but I have found an error in the `sensor.export_yield_sale_2`. Now corrected in the package file. See history log here e5e59a8ab9ea61045f023ecba5e21bd165097f02

![2024-01-27_11-37-39](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/0491bcfb-c4b6-40a0-ac1c-11fa8f9b9e06)

**Comment by JacobsenKim** [comment]
Jeg var lige ved at ? ind til den da jeg ikke forstod hvad der var sket med den sensor.

![Skærmbillede fra 2024-01-27 12-41-46](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/ac28abaf-7ecb-4879-93e5-9252e5befaa2)
![Skærmbillede fra 2024-01-27 12-38-32](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/113f467d-bea7-4381-b91b-b508c796d48e)
![Skærmbillede fra 2024-01-27 12-50-07](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/bb22a2cf-e575-4ff8-9b18-c302dd02c04d)

Nu kommer det store ? så, hvordan får jeg den til den korrekte værdi, jeg ved hvordan man ændre værdien men ikke hvor jeg kan finde den korrekte værdi.

**Reply by JacobsenKim** [reply]
Glemte lige et billede efter koden er rettet og genstart er lavet, så er de alle med DDK

![Skærmbillede fra 2024-01-27 14-00-14](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/dd46e5f4-9093-4317-b540-6dfac318916e)

**Comment by JensenNick** [comment]
@JacobsenKim Purge og Reset Utility Meter inde under Developer Tools > SERVICES. Jeg er dog bange for at der er et problem med de float(0) jeg fik rettet den 22 januar. Se her https://github.com/JensenNick/huawei_solar_pees/issues/13

**Reply by JacobsenKim** [reply]
Jeg har sat den til det samme som de 2 andre sensor da jeg tror det er det samme, men har svært ved at gennemskue lige den sensor da der sker lidt om formning med den.

men nu er den sat til det samme som de 2 første sensor

I /developer-tools/template

sensor.pv_economy_result_w_pv -  {{ ( states('sensor.pv_economy_result_w_pv') | float(0) | round(5)) }}
sensor.yearly_pv_economy_result_w_pv - {{ ( states('sensor.yearly_pv_economy_result_w_pv') | float(0) | round(5)) }}
sensor.export_yield_sale_2 - {{ ( states('sensor.export_yield_sale_2') | float(0) | round(5)) }}

![Skærmbillede fra 2024-01-27 15-54-03](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/1d2ce428-8a6b-4bfb-bf7f-8cb99fb310ec)

Jeg har også justeret Utility Meter på:

sensor.monthly_house_load_energy

sensor.quarterly_house_load_energy

sensor.yearly_house_load_energy

Med:

service: utility_meter.calibrate
data:
  entity_id: sensor.yearly_export_yield_sale
  value: "90.03771"

**Reply by JensenNick** [reply]
@JacobsenKim Når jeg kigger på din graf, så er jeg ikke sikker på at det er et float(0) problem. Et problem med float(0) vil få værdien til at spinge lodret ved gestart o-lig - så tænker at vi skal fortsætte i denne tråd. Det er temelig sikkert ikke den rigtige værdi du har sat den til. De to førstevil være det samme det første år og `sensor.export_yield_sale_2` er det du har solgt produktion for (export yield = eksporteret produktion). Her er lige en henvisning til [Wiki Pages](https://github.com/JensenNick/huawei_solar_pees/wiki), der beskriver de forskellige sensorer.

**Reply by JacobsenKim** [reply]
Nej det her er pga. den har været koblet på en forkert sensor så den har lavet noget kWh.

ang. sensor så må det være dem her den passer sammen med.

Wiki siger: 

> Income related to export of solar PV yield directly ti grid.

sensor.export_yield_sale_2    - {{ ( states('sensor.export_yield_sale_2') | float(0) | round(5)) }}

men er det ikke det samme som 

> Export - Income
sensor.export_income_2	Income related to export of electricity either yield from the solar PV or discharging of the battery to grid.

sensor.pv_economy_income_w_pv - {{ ( states('sensor.pv_economy_income_w_pv') | float(0) | round(5)) }}
yearly                        - {{ ( states('sensor.yearly_pv_economy_income_w_pv') | float(0) | round(5)) }}
sensor.export_income_2        - {{ ( states('sensor.export_income_2') | float(0) | round(5)) }}


Det er lidt (meget) forvirrende at finde rundt i når man ikke selv har lavet det og ikke ved hvorfor der er flere med samme funktion.

**Reply by JensenNick** [reply]
@JacobsenKim Der er ganske få, med samme funktion og hvor der er sensorer med samme funktion, er der gjort opmærksom herpå i Wiki. Hvor det er gjort, er det netop for at kunne bevare overblikket og ikke skulle behøve "krydse" sensorer på tværs af "grupperne". Håber der giver mening. Men enig der er mange, men der er en forklaring og et behov for dem alle.
* `sensor.export_yield_sale_2` fortæller hvor meget du har fået for din eksport af produktion fra solcelleanlægget (yield). Eksporten er opdelt i direkte eksport af produktion (yield) fra solcelleanlæg og afladning af batteriet til net (grid). Årsagen hertil er at eks. i UK er der mange der med stor fortjeneste oplader batteriet billigt og aflader i visse perioder hvor de får op til 4 pund pr. kWh de sende på net.
* `sensor.pv_economy_income_w_pv` fortæller hvor meget du har solgt for, og ja det er rigtigt nok samme `sensor.export_income_2`, men for at kunne bevare overblikket.
